content/docs/scripting.org

```diff
@@ -0,0 +1,180 @@
+Again, we don't have to worry about allocation and deallocating memory. Moreover, string maniuplation are way easier in Python.
+#+BEGIN_SRC python
+new_name = sys.argv[1] + ": " + sys.argv[2]
+#+END_SRC
+
+*** Part 4: Getting the current name of the workspace from the id of the workspace we want to rename
+There is no std regex functions or anything that contains string manipulations in Zig, so we have to do it by hand by analyzing the characters of the string. Obviously, this is a very error-prone and hard task.
+#+BEGIN_SRC zig
+const needle: []const u8 = try std.fmt.allocPrint(allocator, "num\":{s}", .{args[1]});
+defer allocator.free(needle);
+var pos_in_text_result = std.mem.indexOf(u8, text_result, needle);
+pos_in_text_result.? += 15;
+if (std.mem.eql(u8, args[1], "10")) {
+    pos_in_text_result.? += 1;
+}
+
+var start_of_name = std.mem.indexOf(u8, text_result[pos_in_text_result.? .. pos_in_text_result.? + 50], ",\"");
+start_of_name.? += 2;
+start_of_name.? += pos_in_text_result.?;
+
+var end_of_name = std.mem.indexOf(u8, text_result[pos_in_text_result.? .. pos_in_text_result.? + 50], "\"");
+end_of_name.? += start_of_name.?;
+
+const diff = end_of_name.? - start_of_name.?;
+const name = text_result[start_of_name.? - 3 - diff .. end_of_name.? - 3 - diff];
+const old_name = try std.fmt.allocPrint(allocator, "\"{s}\"", .{name});
+defer allocator.free(old_name);
+#+END_SRC
+In Python, we the help of a regex library we can easily extract the name of the workspace from the json output.
+#+BEGIN_SRC python
+pattern = fr'"num":{sys.argv[1]},"name":"([^"]+)"'
+match = re.search(pattern, result.stdout)
+
+if match:
+    old_name = match.group(1)
+
+print(old_name)
+#+END_SRC
+
+*** Part 5: Re running a process to rename the workspace
+Finally, we are going to run the process to rename the workspace using the inputs we have extracted and created.
+Here we arrive at the same conclusions that we had in Part 2
+#+BEGIN_SRC zig
+params = &.{ app_name, "rename", "workspace", old_name, "to", new_name };
+
+const modif_result = try std.ChildProcess.run(.{
+    .allocator = allocator,
+    .argv = params,
+    .cwd = null,
+    .cwd_dir = null,
+    .env_map = null,
+    .max_output_bytes = 51200,
+    .expand_arg0 = .no_expand,
+});
+defer {
+    allocator.free(modif_result.stdout);
+    allocator.free(modif_result.stderr);
+}
+#+END_SRC
+And again in Python.
+#+BEGIN_SRC python
+result = subprocess.run(params, shell=True, capture_output=True, text=True)
+#+END_SRC
+
+*** Part 6: Printing the result message
+Here we are going to have to format our string in order to print it.
+#+BEGIN_SRC zig
+std.debug.print("Should have changed workspace {s} from {s} -> {s}", .{ args[1], name, new_name });
+#+END_SRC
+In Python, we can just print the result without the need for special formatting, which is way faster and nicer.
+#+BEGIN_SRC python
+print("Should have changed the name of the workspace with id " + sys.argv[1] + "from" + old_name + " to " + new_name)
```

Comment:
Why you dont use f string:
```py
my_var = 3
print(f"my var : {my_var}")
```

Reply:
could have, does it really change something tho ?